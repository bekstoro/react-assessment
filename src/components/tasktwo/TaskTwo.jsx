import React, { useState } from 'react';

import {
  BlueButton,
  Content,
  GreenButton,
  Heading,
  RedButton,
  Task,
} from '@components/styled-components/Task';
import ConfirmationModal from '@components/modal/ConfirmationModal';
import LoadingModal from '@components/modal/LoadingModal';
import ConfirmationExtendedModal from '@components/modal/ConfirmationExtendedModal';
import { CONFIRMATION_DETAILS } from '@constants';

const TaskTwo = () => {
  const [showConfirmationModal, setShowConfirmationModal] = useState(false);
  const [showConfirmationExtendedModal, setShowConfirmationExtendedModal] = useState(false);
  const [showLoadingModal, setShowLoadingModal] = useState(false);

  const confirmModal = () => {
    console.log('Files are deleted');
  };

  return (
    <Task>
      <Heading>Task Two</Heading>
      <Content>
        <RedButton onClick={() => setShowLoadingModal(true)}>
          open loading modal
        </RedButton>
        <GreenButton onClick={() => setShowConfirmationModal(true)}>
          open confirmation modal
        </GreenButton>
        <BlueButton onClick={() => setShowConfirmationExtendedModal(true)}>
          open confirmation extended modal
        </BlueButton>
      </Content>
      <LoadingModal
        isOpen={showLoadingModal}
        closeModal={() => setShowLoadingModal(false)}
      />
      <ConfirmationModal
        cancellationButtonText="No"
        confirmationButtonText="Yes"
        isOpen={showConfirmationModal}
        message="This action cannot be undone"
        title="Are you sure you want to delete all of your files?"
        closeModal={() => setShowConfirmationModal(false)}
        confirmModal={confirmModal}
      />
      <ConfirmationExtendedModal
        cancellationButtonText="Cancel"
        confirmationButtonText="Delete all"
        isOpen={showConfirmationExtendedModal}
        message={(
          <>
            <p>
              If you delete the
              &nbsp;
              <b>Executive metrics</b>
              &nbsp;
              report, you will also delete the associated history:
            </p>
            <textarea defaultValue={CONFIRMATION_DETAILS} readOnly />
            <p>
              Please type the word &apos;Delete&apos; to remove the
              &nbsp;
              <b>Executive metrics</b>
              &nbsp;
              report and its associated history:
            </p>
          </>
        )}
        title="Are you sure you want to delete this report and its history?"
        closeModal={() => setShowConfirmationExtendedModal(false)}
        confirmModal={confirmModal}
      />
    </Task>
  );
};

export default TaskTwo;
