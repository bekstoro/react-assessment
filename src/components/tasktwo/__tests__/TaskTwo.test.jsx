import React from 'react';
import { render } from '@testing-library/react';

import TaskTwo from '@components/tasktwo/TaskTwo';

describe('TaskTwo', () => {
  it('TaskTwo renders correctly', () => {
    const { container } = render(<TaskTwo />);
    expect(container).toMatchSnapshot();
  });
});
