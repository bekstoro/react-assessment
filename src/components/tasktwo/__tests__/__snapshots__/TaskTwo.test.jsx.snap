// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTwo TaskTwo renders correctly 1`] = `
<div>
  <div
    class="Task-sc-7z955n-0 dBJrLg"
  >
    <h1
      class="Task__Heading-sc-7z955n-1 hiEtuP"
    >
      Task Two
    </h1>
    <section
      class="Task__Content-sc-7z955n-6 fejPFW"
    >
      <button
        class="Task__Button-sc-7z955n-2 Task__RedButton-sc-7z955n-3 joftDn kWdQeZ"
      >
        open loading modal
      </button>
      <button
        class="Task__Button-sc-7z955n-2 Task__GreenButton-sc-7z955n-4 joftDn ciWuGS"
      >
        open confirmation modal
      </button>
      <button
        class="Task__Button-sc-7z955n-2 Task__BlueButton-sc-7z955n-5 joftDn iEvgG"
      >
        open confirmation extended modal
      </button>
    </section>
  </div>
</div>
`;
